cleverhans/experimental/certification/optimization.py

```diff
@@ -108,6 +109,61 @@ def _vector_prod_fn(x):
         vector_prod_fn=_vector_prod_fn)
     return estimated_eigen_vector
 
+  def construct_lanczos_params(self, k):
+    """Constructs TF ops for storing Lanczos matrices.
+
+    Args:
+      k: number of iterations and dimensionality of the tridiagonal matrix
+    """
+    # Using autograph to automatically handle
+    # the control flow of minimum_eigen_vector
+    self.lanczos_eigen_vec = autograph.to_graph(utils.lanczos_decomp)
+
+    def _vector_prod_fn(x):
+      return self.dual_object.get_psd_product(x)
+
+    # First, get the eigenvalue corresponding to largest magnitude
+    self.alpha, self.beta, self.W = self.lanczos_eigen_vec(_vector_prod_fn,
+                                                           0,
+                                                           self.dual_object.matrix_m_dimension,
+                                                           MIN_LANCZOS_ITER)
+    self.eig_max_placeholder = tf.placeholder(tf.float32, shape=[])
+
+    # Shift the matrix by the largest magnitude eigenvalue to compute
+    # the smallest real eigenvalue
+    self.alpha_hat, self.beta_hat, self.W_hat = self.lanczos_eigen_vec(_vector_prod_fn,
+                                                                       self.eig_max_placeholder,
+                                                                       self.dual_object.matrix_m_dimension,
+                                                                       k)
+
+  def get_lanczos_eig_vec(self):
+    """Computes the min eigen value and corresponding vector of matrix M
+    using the Lanczos algorithm.
+
+    Returns:
+      eig_vec: Corresponding eigen vector
+      eig_val: Minimum absolute eigen value
+    """
+    alpha, beta, _ = self.sess.run([self.alpha, self.beta, self.W])
```

Comment:
computation of W does not have any side effects (i.e. changing variables, etc...), so you can just do following:
```
alpha, beta = sess.run([self.alpha, self.beta])
```